providers/fips/self_test_data.inc

```diff
@@ -8,18 +8,18 @@
  */
 
 /* Macros to build Self test data */
-#define ITM(x) x, sizeof(x)
-#define ITM_STR(x) x, (sizeof(x) - 1)
+#define ITM(x) ((void *)&x), sizeof(x)
+#define ITM_STR(x) ((void *)&x), (sizeof(x) - 1)
 
 #define ST_KAT_PARAM_END() { "", 0, NULL, 0 }
 #define ST_KAT_PARAM_BIGNUM(name, data)                                        \
     { name, OSSL_PARAM_UNSIGNED_INTEGER, ITM(data) }
 #define ST_KAT_PARAM_OCTET(name, data)                                         \
     { name, OSSL_PARAM_OCTET_STRING, ITM(data) }
 #define ST_KAT_PARAM_UTF8STRING(name, data)                                    \
-    { name, OSSL_PARAM_UTF8_STRING, ITM_STR(data) }
+    { name, OSSL_PARAM_UTF8_STRING, ITM(data) }
```

Comment:
This is weird change, because ST_KAT_PARAM_UTF8STRING is used with C string literals elsewhere in the file. So better just not use it for the sshkdf_type case.

Reply:
So...  you'd rather I have it like this?

``` C
    { name, OSSL_PARAM_UTF8_STRING, &data, sizeof(data) }
```

Reply:
Ah, never mind, now I see what you're saying...

Reply:
I think I've fixed it